packages/analytics-js/src/components/eventManager/utilities.ts

```diff
@@ -269,18 +268,21 @@ const getEnrichedEvent = (
   }
 
   if (rudderEvent.type === RudderEventType.Identify) {
-    (commonEventData.context as RudderContext).traits = traits
-      ? mergeDeepRight(traits, clone(state.session.userTraits.value) as ApiObject)
+    (commonEventData.context as RudderContext).traits = rudderEvent.context?.traits
+      ? mergeDeepRight(
+          rudderEvent.context?.traits,
+          clone(state.session.userTraits.value) as ApiObject,
+        )
```

Comment:
Let's use the storage type to decide whether to the API or state data.

Reply:
Addressed